core/templates/dev/head/pages/exploration_editor/improvements_tab/ImprovementsTabDirective.js

```diff
@@ -26,37 +26,26 @@ oppia.directive('improvementsTab', [
         '/pages/exploration_editor/improvements_tab/' +
         'improvements_tab_directive.html'),
       controller: [
-        '$scope', function($scope) {
-          // NOTE: The constants used in this controller are intentionally
-          // hard-coded to demonstrate that the Improvements Tab is configured
-          // correctly. The numbers are arbitrary, and were simply copied from
-          // the [design doc](https://bit.ly/2Bp7YLS).
+        '$scope', 'ImprovementCardService',
+        function($scope, ImprovementCardService) {
+          var fetchedCards = [];
+          ImprovementCardService.fetchCards().then(function(cards) {
+            fetchedCards = cards;
+          });
 
-          $scope.resolvedIssueCount = 13;
-          $scope.archivedIssueCount = 30;
-
-          var issueTypeMaterialIconMapping = {
-            feedback: 'feedback',
-            playthrough: 'gamepad',
-            unresolved_answer: 'help',
+          $scope.getCards = function() {
+            return fetchedCards;
           };
-
-          $scope.issueTypeToMaterialIcon = function(issueType) {
-            if (issueTypeMaterialIconMapping.hasOwnProperty(issueType)) {
-              return issueTypeMaterialIconMapping[issueType];
-            } else {
-              return '';
-            }
+          $scope.getResolvedCardsCount = function() {
+            return fetchedCards.reduce(function(resolvedCardsCount, card) {
```

Comment:
Optional: perhaps .filter().length is more readable? Ditto below.

Also, missing tests?

Reply:
Done.